### ISSUE TITLE
The asynchronous training hangs.

### ISSUE THREAD

**Issue by jianhai0527** [verl-project/verl#5815]
### System Info

verl 0.8.0dev main分支

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

运行main分支的异步训练脚本：verl/experimental/fully_async_policy/shell/dapo_7b_math_fsdp2_4_4.sh

卡死了
<img width="1465" height="47" alt="Image" src="https://github.com/user-attachments/assets/2f8cd258-5ba2-4630-8a67-d6564732dbaf" />

### Expected behavior

non

**Comment by Kirrito-k423** [comment]
To diagnose the asynchronous training hang, more information is needed:

1. **Training stage where it hangs**: Is it during:
   - Rollout generation?
   - Reward computation?
   - Parameter update?
   - Weight synchronization?

2. **Log output**: What was the last log line before the hang? Check Ray actor logs for any blocking calls.

3. **Resource utilization**: Check `nvidia-smi` or `htop` for:
   - GPU memory usage (potential OOM)
   - CPU utilization (potential deadlock)

4. **Network/NCCL**: If multi-node, check NCCL communication:
   ```bash
   export NCCL_DEBUG=INFO
   export NCCL_BLOCKING_WAIT=1
   ```

5. **Async training config**: Key parameters that affect behavior:
   - `async_training.staleness_threshold`
   - `async_training.trigger_parameter_sync_step`
   - `async_training.require_batches`

Please run with `trainer.logger=[console]` and share the last 20-30 log lines before the hang. Also check `dmesg` for any kernel-level errors (OOM kills, GPU errors).

**Comment by jianhai0527** [comment]
cross cluster]
(WorkerDict pid=76630) [dsw-720814-6bf49444f8-mm5mw][76630:77518][0][init.cc:1335] NCCL INFO Trees [0] 2/-1/-1->1->0 [1] 2/-1/-1->1->0 [2] 2/-1/-1->1->0 [3] 2/-1/-1->1->0 [4] -1/-1/-1->1->2 [5] -1/-1/-1->1->2 [6] -1/-1/-1->1->2 [7] -1/-1/-1->1->2 [8] 5/-1/-1->1->3 [9] 5/-1/-1->1->3 [10] 3/-1/-1->1->5 [11] 3/-1/-1->1->5 [12] 2/-1/-1->1->0 [13] 2/-1/-1->1->0 [14] 2/-1/-1->1->0 [15] 2/-1/-1->1->0 [16] -1/-1/-1->1->2 [17] -1/-1/-1->1->2 [18] -1/-1/-1->1->2 [19] -1/-1/-1->1->2 [20] 5/-1/-1->1->3 [21] 5/-1/-1->1->3 [22] 3/-1/-1->1->5 [23] 3/-1/-1->1->5 [repeated 7x across cluster]
(WorkerDict pid=76634) [dsw-720814-6bf49444f8-mm5mw][76634:77520][0][topo.cc:1104] NCCL INFO Setting affinity for GPU 5 to ffffffff [repeated 3x across cluster]
(WorkerDict pid=76630) [dsw-720814-6bf49444f8-mm5mw][76630:77518][0][paths.cc:888] NCCL INFO 24 coll channels, 0 fc channels, 32 p2p channels, 4 p2p channels per peer [repeated 7x across cluster]
(WorkerDict pid=76630) [dsw-720814-6bf49444f8-mm5mw][76630:77518][0][init.cc:1395] NCCL INFO Begin connect topo graphs [repeated 7x across cluster]
(WorkerDict pid=76630) [dsw-720814-6bf49444f8-mm5mw][76630:77557][0][proxy.cc:1418] NCCL INFO [Proxy Service UDS] Device 0 CPU core 47 [repeated 7x across cluster]
(WorkerDict pid=76635) [dsw-720814-6bf49444f8-mm5mw][76635:77521][0][p2p.cc:381] NCCL INFO Channel 17 : 6[15000] -> 5[1c000] via P2P/IPC pointer intra-node [repeated 392x across cluster]
(WorkerDict pid=76630) [dsw-720814-6bf49444f8-mm5mw][76630:77518][0][topo.cc:1104] NCCL INFO Setting affinity for GPU 1 to ffffffff,00000000 [repeated 3x across cluster]
(WorkerDict pid=76630) [dsw-720814-6bf49444f8-mm5mw][76630:77518][0][generic.cc:28] NCCL INFO Connected all rings [repeated 7x across cluster]
(WorkerDict pid=76630) [dsw-720814-6bf49444f8-mm5mw][76630:77518][0][generic.cc:54] NCCL INFO Connected all trees [repeated 7x across cluster]
(WorkerDict pid=76633) [dsw-720814-6bf49444f8-mm5mw][76633:77519][0][ext_kernel.cc:141] NCCL INFO ExtKernel/Plugin: Using extKernel plugin CustomAllReduce [repeated 7x across cluster]
(WorkerDict pid=76630) [dsw-720814-6bf49444f8-mm5mw][76630:77518][0][tuning.cc:555] NCCL INFO threadThresholds 8/8/64/64/64/64/64 | 64/8/64/64/64/64/64 | 8/8/64/64/64/64/64 | 8/8/64/64/64/64/64 | 8/8/64/64/64/64/64 [repeated 7x across cluster]
(WorkerDict pid=76630) [dsw-720814-6bf49444f8-mm5mw][76630:77518][0][init.cc:1711] NCCL INFO ncclCommInitRank comm 0x7efc6590fb40 rank 1 nranks 8 hggcDev 0 busId 3d000 commHash 317500fb490cac5f maxConnType 1 - Init COMPLETE [repeated 7x across cluster]
(FullyAsyncRollouter pid=77568) dataset len: 1791700
(FullyAsyncRollouter pid=77568) Using dataset class: RLHFDataset
(FullyAsyncRollouter pid=77568) dataset len: 960
(FullyAsyncRollouter pid=77568) [FullyAsyncRollouter] Rollouter _create_dataloader...
(FullyAsyncRollouter pid=77568) <verl.utils.dataset.rl_dataset.RLHFDataset object at 0x7f88c040ee10>
(FullyAsyncRollouter pid=77568) <verl.utils.dataset.rl_dataset.RLHFDataset object at 0x7f88c07c7620>
(FullyAsyncRollouter pid=77568) Size of train dataloader: 1791700, Size of val dataloader: 1
(FullyAsyncRollouter pid=77568) Total training steps: 17917000
(FullyAsyncRollouter pid=77568) [FullyAsyncRollouter] Total rollout steps: 51200
(FullyAsyncRollouter pid=77568) W0402 14:23:49.867000 77568 site-packages/torch/utils/cpp_extension.py:117] No CUDA runtime is found, using CUDA_HOME='/usr/local/PPU_SDK/CUDA_SDK'
(WorkerDict pid=76629) Flash Attention 2 only supports torch.float16 and torch.bfloat16 dtypes, but the current dype in Qwen2Model is torch.float32. You should run training or inference using Automatic Mixed-Precision via the `with torch.autocast(device_type='torch_device'):` decorator, or load the model with the `dtype` argument. Example: `model = AutoModel.from_pretrained("openai/whisper-tiny", attn_implementation="flash_attention_2", dtype=torch.float16)` [repeated 14x across cluster]
Loading weights:   0%|                                                                                                                          | 0/290 [00:00<?, ?it/s] [repeated 7x across cluster]
Loading weights:   0%|▍                                                                                                                 | 1/290 [00:00<00:41,  6.93it/s] [repeated 5x across cluster]
Loading weights: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████| 290/290 [00:00<00:00, 1433.55it/s]
Loading weights: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████| 290/290 [00:00<00:00, 1379.63it/s] [repeated 5x across cluster]
(FullyAsyncRollouter pid=77568) benchmark_serving: failed to import start_active_profile
(FullyAsyncRollouter pid=77568) Using blocking ray.get inside async actor. This blocks the event loop. Please use `await` on object ref with asyncio.gather if you want to yield execution to the event loop instead.
(CheckpointEngineWorker pid=78250) [Gloo] Rank 0 is connected to 0 peer ranks. Expected number of connected peer ranks is : 0

------
1. here are the last 20-30 log lines before the hang
2. I've already checked, and there are no OOM or other abnormalities.

**Comment by jianhai0527** [comment]
Total Usage:
 35.0/64.0 CPU (14.0 used of 42.0 reserved in placement groups)
 4.666200000000001/16.0 GPU (4.666200000000001 used of 14.0 reserved in placement groups)
 0B/1008.71GiB memory
 0B/186.26GiB object_store_memory

Total Constraints:
 (no request_resources() constraints)
Total Demands:
 {'CPU': 3.0, 'GPU': 1.0} * 1 (PACK): 1+ pending placement groups

-----
here is the log from ray status
It seems to be waiting for resources, but there are enough resources available.
